### DISCUSSION TITLE
I want to use Google.OrTools for .NET 7.0, how can I build it?

### DISCUSSION THREAD

**Discussion by hexianggui** [google/or-tools#3867] [Build and configuration issues]
ll of my projects are in .NET 7.0, but I have a NuGet package that is designed for 6.0. Occasionally, I encounter an error: 'Could not load file or assembly 'Google.OrTools, Version=9.6.2534.0, Culture=neutral, PublicKeyToken=37e4c4e778bf1216'

**Comment by CADBIMDeveloper** [comment]
Just for the information, I have no problems to build an application for .NET 7 which is using the latest ortools (9.6) it works for me both on Windows (my dev machine) and in the cloud inside a docker container on Linux machine

**Comment by hexianggui** [comment]
Yes, initially, I had no issues when running the application. However, recently, after the first few calls or so, I started encountering an error: "Could not load file or assembly 'Google.OrTools, Version=9.6.2534.0, Culture=neutral, PublicKeyToken=37e4c4e778bf1216'." I'm not sure of the exact reason behind this

**Comment by Mizux** [comment]
strange....
My 2 cents,
1. OR-Tools is a native package so basically you have a package for each RID supported
```sh
unzip -l google.ortools.runtime.win-x64.9.6.2534.nupkg
Archive:  google.ortools.runtime.win-x64.9.6.2534.nupkg
  Length      Date    Time    Name
---------  ---------- -----   ----
      524  2023-03-10 12:05   _rels/.rels
     1045  2023-03-10 12:05   Google.OrTools.runtime.win-x64.nuspec
 19580416  2023-03-10 12:05   runtimes/win-x64/native/google-ortools-native.dll
```
and the "meta" package agnostic of the platform but targeting few TFM
```sh
unzip -l google.ortools.9.6.2534.nupkg
Archive:  google.ortools.9.6.2534.nupkg
  Length      Date    Time    Name
---------  ---------- -----   ----
      508  2023-03-13 10:58   _rels/.rels
     2286  2023-03-13 10:58   Google.OrTools.nuspec
  1670144  2023-03-13 10:58   lib/net46/Google.OrTools.dll
   773630  2023-03-13 10:58   lib/net46/Google.OrTools.xml
  1670144  2023-03-13 10:58   lib/net6.0/Google.OrTools.dll
   773630  2023-03-13 10:58   lib/net6.0/Google.OrTools.xml
   ...
```

On windows IIRC I use VS 2022 Community to build so you need to redistributable associated to be able to load the `runtimes/win-x64/native/google-ortools-native.dll`

ps: My play sandbox https://github.com/mizux/dotnet-native
https://github.com/Mizux/dotnet-native/actions/workflows/amd64_tfm.yml
